### DISCUSSION TITLE
Better Stage + Schedule Integration

### DISCUSSION THREAD

**Discussion by rashfael** [venueless/venueless#257] [Feature Preview & Feedback]
Currently the event schedule is pretty isolated from the rest of venueless, particulary stages, where most talks are being held.
As an attendee it would be nice to see at a glance, which talks/sessions are currently happening in which stages and which sessions are upcoming.

A while ago we built a prototype how integrating the event schedule into stages could look like:

# Attendee-facing schedule display
Apart from the obvious enhancements like showing the session title in the stage title bar, we are also experimenting with showing the currently running session for a stage directly in the sidebar and building a new landing page with a list of upcoming sessions.
![2021-07-14-20-04-03](https://user-images.githubusercontent.com/1093590/125670880-321fd7a3-306d-4718-a770-4dab37fe95d9.png)

## Current sessions in sidebar
![2021-07-14-20-04-33](https://user-images.githubusercontent.com/1093590/125671238-f2febfd0-b9a7-45d8-bb01-bfd854fc7a32.png)
For showing schedule information directly in the sidebar we looked at how attendees are finding the content they want to watch and interact with. We believe that the actual session information matters more than the stage name, so for stages with an associated schedule session we emphasized the session name and speaker(s) and the schedule state (live or soon) at the cost of the stage name itself.

## Immediate schedule in landing page
![2021-07-14-20-04-43](https://user-images.githubusercontent.com/1093590/125671835-94c82d9d-63fa-4fed-8394-08c7c2c7792c.png)
We think showing the next X sessions from the schedule directly on the landing page is a good idea, so attendees don't have to look inside the full schedule. So we built it :wink:
This list could also be combined with an attendees individual schedule (which is also on our long list of improvements).

# Organizer-facing features
To show attendees the current schedule information it has be set *somehow*.
While just advancing the schedule strictly and automatically by the clock might seem the obvious solution, we quickly saw some problems with it:
- You would be unable to end or start a session late.
- Due to how schedules are distributed, any last minute changes might not be available to all attendees.
- No short breaks which aren't in the schedule.
- Probably a lot of edge cases more we did not consider

Instead we opted to let organizers manually control which session is the current one for a stage. This currently looks something like this:
![2021-07-14-20-05-02](https://user-images.githubusercontent.com/1093590/125672877-8eb62450-871e-4be6-a4bf-2527e490e500.png)

Right now, moderators or anyone with the right permission can pick a session for a stage from a list and set it as the current one.
This view has a lot of room for improvements, we already thought of:
- Simple "next scheduled" button
- "break" button
- switchable autopilot mode
- freeform inputs if you want to go off script

**Have any questions, comments or ideas for the previewed features?
What schedule information would you like to see at a glance?
How common are images for sessions / speakers in your events and do you feel adding them to the sidebar and landing page sessions would improve usability?
Are you an event organizer and are interested in trying out this new feature?
Comment below!**

**Comment by raphaelm** [comment]
Unlike when we first discussed this a long time ago, I think a opt-in or even opt-out autopilot mode would be a must-have for most events. When I look into running events, they are surprisingly on-time. Also, most event teams are heavily understaffed. For a stage, they often need two people already, one doing the live video mixing in StreamYard/OBS, and one monitoring the chat and questions, plus maybe one giving tech support to attendees, which is already a lot for them, especially if its done by unexperienced people.

**Reply by Aelindara** [reply]
Upvote fot "opt-in or even opt-out autopilot mode", or something similar to a pause autopilot button.

**Comment by Aelindara** [comment]
I like showing the current session in the sidebar. It gives the whole UI are more dynamic and up to date feel. And it's also helpfull for attendees to see what is happening where right now, without the need to switch between views.

In my opinion showing the title and the name of the speaker (or speakers) is all the information you need to know as an attendee. Because most people go by "I want to see the talk about XYZ" or "I want to see speaker ABC. They were great last time".  I wouldn't add images in the sidebar, too much going on. In the landing page it could work more as an nice design option than an usability thing. Easier to judge with a mockup what it could look like.

**Comment by tthePB** [comment]
Great feature! From my organizer perspective this would be very helpful and also makes the landing page much more useful. I also agree on the opt-in/opt-out feature for the autopilot.

**Comment by saschafoerster** [comment]
I also think that spontaneous changes of the schedule don't happen so often. Is this is likely to happen, most of the times we use pretalx and we change the timings in there. So in 99% of our cases we just need to see and show, what was planned, be it planned with a excel sheet or with pretalx.
Some "general message" or "announcement" that reaches everyone, of course would make it possible to announce little delays and even more.

**Comment by rashfael** [comment]
Following your feedback we focused more on the "autopilot" version of this feature, which is now available to test behind the `schedule-control` feature flag. If you are self-hosting, you can enable this flag for yourself, if you are using venueless hosted on venueless.org, you can contact our support to unlock this feature for your events.

Enabling the `schedule-control` feature will show a `Schedule` panel with a switch `Compute current session from schedule` in the room manage view.

![2021-08-31-15-35-42](https://user-images.githubusercontent.com/1093590/131512386-8f8d9322-5cab-4e62-8bec-951a63962134.png)

With this switch you can enable automatic schedule integration per room, which will show session title and speakers in the sidebar and room header.

We'll continue to develop schedule integration and will add more features, but you can use this fully automatic version right now.

A demo event with a special schedule to see sessions change more quickly than in a real event is also planned.

**Comment by MrGamy** [comment]
Initial customer feedback:

> We expected the setting around here [in the room settings] instead.
> <img width="364" alt="Sidebar addons menu in room settings" src="https://user-images.githubusercontent.com/8639512/132674189-fe549521-11f6-4c32-b7d9-3e109c453100.png">
